### ISSUE TITLE
panic in handler

### ISSUE THREAD

**Issue by djdv** [ipfs/go-ipfs#4823]
682b263f3a54ffa36409422d922e80adead87e0c

~~I'm not sure what triggered this, it may have been when opening a hash via the gateway.~~ I'm guessing `/core/commands/publish.go:96` is the problem point causing an out of range but it could be elsewhere here.

```
Daemon is ready
18:55:51.133 ERROR  providers: base32 decoding error: illegal base32 data at input byte 55 providers.go:102
...
07:55:37.541 ERROR  providers: base32 decoding error: illegal base32 data at input byte 55 providers.go:102
08:32:43.324 ERROR  cmds/http: a panic has occurred in the commands handler! asm_amd64.s:573
08:32:43.326 ERROR  cmds/http: runtime error: index out of range asm_amd64.s:573
08:32:43.327 ERROR  cmds/http: stack trace:
goroutine 56483732 [running]:
runtime/debug.Stack(0xc0422269a0, 0xc0d0070f50, 0x1)
        C:/Go/src/runtime/debug/stack.go:24 +0xae
gx/ipfs/QmabLouZTZwhfALuBcssPvkzhbYGMb4394huT7HY4LQ6d3/go-ipfs-cmds/http.(*handler).ServeHTTP.func1()
        C:/Users/Dominic/Projects/Go/src/gx/ipfs/QmabLouZTZwhfALuBcssPvkzhbYGMb4394huT7HY4LQ6d3/go-ipfs-cmds/http/handler.go:94 +0x130
panic(0xcef380, 0x1548c00)
        C:/Go/src/runtime/panic.go:505 +0x237
github.com/ipfs/go-ipfs/core/commands.glob..func96(0x1002d40, 0xc0c65ad2e0, 0x1001b20, 0xc0a851ec30)
        C:/Users/Dominic/Projects/Go/src/github.com/ipfs/go-ipfs/core/commands/publish.go:96 +0x7a5
github.com/ipfs/go-ipfs/commands/legacy.NewCommand.func1(0xc0c5dd3dc0, 0x4f0f84d0, 0xc0c5dd3e30, 0xfed4e0, 0xc0a721ac40)
        C:/Users/Dominic/Projects/Go/src/github.com/ipfs/go-ipfs/commands/legacy/command.go:38 +0x1be
gx/ipfs/QmabLouZTZwhfALuBcssPvkzhbYGMb4394huT7HY4LQ6d3/go-ipfs-cmds.(*Command).Call(0x1557580, 0xc0c5dd3dc0, 0x4f0f84d0, 0xc0c5dd3e30, 0xfed4e0, 0xc0a721ac40)
        C:/Users/Dominic/Projects/Go/src/gx/ipfs/QmabLouZTZwhfALuBcssPvkzhbYGMb4394huT7HY4LQ6d3/go-ipfs-cmds/command.go:106 +0x24d
gx/ipfs/QmabLouZTZwhfALuBcssPvkzhbYGMb4394huT7HY4LQ6d3/go-ipfs-cmds/http.(*handler).ServeHTTP(0xc09a1aa9a0, 0xff6c40, 0xc0b5344518, 0xc0c49d0d00)
        C:/Users/Dominic/Projects/Go/src/gx/ipfs/QmabLouZTZwhfALuBcssPvkzhbYGMb4394huT7HY4LQ6d3/go-ipfs-cmds/http/handler.go:160 +0x8d9
gx/ipfs/QmabLouZTZwhfALuBcssPvkzhbYGMb4394huT7HY4LQ6d3/go-ipfs-cmds/http.prefixHandler.ServeHTTP(0xe1154f, 0x7, 0xfedea0, 0xc09a1aa9a0, 0xff6c40, 0xc0b5344518, 0xc0c49d0d00)
        C:/Users/Dominic/Projects/Go/src/gx/ipfs/QmabLouZTZwhfALuBcssPvkzhbYGMb4394huT7HY4LQ6d3/go-ipfs-cmds/http/apiprefix.go:24 +0xcf
gx/ipfs/QmPG2kW5t27LuHgHnvhUwbHCNHAt2eUcb4gPHqofrESUdB/cors.(*Cors).Handler.func1(0xff6c40, 0xc0b5344518, 0xc0c49d0d00)
        C:/Users/Dominic/Projects/Go/src/gx/ipfs/QmPG2kW5t27LuHgHnvhUwbHCNHAt2eUcb4gPHqofrESUdB/cors/cors.go:188 +0x105
net/http.HandlerFunc.ServeHTTP(0xc09a1aa9e0, 0xff6c40, 0xc0b5344518, 0xc0c49d0d00)
        C:/Go/src/net/http/server.go:1947 +0x4b
net/http.(*ServeMux).ServeHTTP(0xc07ccfa540, 0xff6c40, 0xc0b5344518, 0xc0c49d0d00)
        C:/Go/src/net/http/server.go:2337 +0x137
github.com/ipfs/go-ipfs/core/corehttp.CheckVersionOption.func1.1(0xff6c40, 0xc0b5344518, 0xc0c49d0d00)
        C:/Users/Dominic/Projects/Go/src/github.com/ipfs/go-ipfs/core/corehttp/commands.go:175 +0x129
net/http.HandlerFunc.ServeHTTP(0xc09a1aa920, 0xff6c40, 0xc0b5344518, 0xc0c49d0d00)
        C:/Go/src/net/http/server.go:1947 +0x4b
net/http.(*ServeMux).ServeHTTP(0xc0421f9440, 0xff6c40, 0xc0b5344518, 0xc0c49d0d00)
        C:/Go/src/net/http/server.go:2337 +0x137
net/http.(Handler).ServeHTTP-fm(0xff6c40, 0xc0b5344518, 0xc0c49d0d00)
        C:/Go/src/net/http/h2_bundle.go:5475 +0x54
gx/ipfs/QmX3QZ5jHEPidwUrymXV1iSCSUhdGxj15sm2gP4jKMef7B/client_golang/prometheus.InstrumentHandlerFuncWithOpts.func1(0xff77c0, 0xc0dc447ce0, 0xc0c49d0d00)
        C:/Users/Dominic/Projects/Go/src/gx/ipfs/QmX3QZ5jHEPidwUrymXV1iSCSUhdGxj15sm2gP4jKMef7B/client_golang/prometheus/http.go:287 +0x276
net/http.HandlerFunc.ServeHTTP(0xc07ce688c0, 0xff77c0, 0xc0dc447ce0, 0xc0c49d0d00)
        C:/Go/src/net/http/server.go:1947 +0x4b
net/http.(*ServeMux).ServeHTTP(0xc0421f92c0, 0xff77c0, 0xc0dc447ce0, 0xc0c49d0d00)
        C:/Go/src/net/http/server.go:2337 +0x137
net/http.serverHandler.ServeHTTP(0xc07d140000, 0xff77c0, 0xc0dc447ce0, 0xc0c49d0d00)
        C:/Go/src/net/http/server.go:2694 +0xc3
net/http.(*conn).serve(0xc0c7488f00, 0xff8a40, 0xc0a8873b80)
        C:/Go/src/net/http/server.go:1830 +0x658
created by net/http.(*Server).Serve
        C:/Go/src/net/http/server.go:2795 +0x282
 asm_amd64.s:573
08:55:46.372 ERROR  providers: base32 decoding error: illegal base32 data at input byte 55 providers.go:102
```

**Comment by Stebalien** [comment]
Damn... I know what's happening. See: https://github.com/ipfs/go-ipfs-cmds/issues/58.

**Comment by Stebalien** [comment]
So, looking into this, I don't think we really *can* fix these bugs properly due to https://github.com/ipfs/go-ipfs-cmds/issues/77.

**Comment by whyrusleeping** [comment]
@djdv is it possible you were messing around with enabling stdin support on windows when this happened? I can see how this might happen if that were the case.

**Comment by djdv** [comment]
@whyrusleeping 
Nailed it.
I can instigate this by doing `ipfs object get` followed by Ctrl-C to cancel/kill while it's waiting for stdin, this incites panic in the daemon and prints this on the console that initiated the request:
`Post http://127.0.0.1:5001/api/v0/object/get?encoding=json&stream-channels=true: context canceled`

Despite being the master branch I must have built it with the `go-ipfs-cmds` lib symlinked, sorry for the confusion.

**Comment by whyrusleeping** [comment]
@djdv no worries, its still an issue. We can reproduce this on linux by running `ipfs name publish` then pressing ctrl+d

The fix is going to be pretty annoying, but we're working on it.

**Comment by Stebalien** [comment]
This can still be triggered on master by running `ipfs name publish` and hitting `^D`.

**Comment by Stebalien** [comment]
I'll make a PR to fix this once #4825 and #4826 have been merged (the "delete dead code commit" allows me to avoid updating that dead code).
